app/src/lib/editors/win32.ts

```diff
@@ -705,147 +637,28 @@ async function findApplication(editor: ExternalEditor): Promise<string | null> {
  */
 export async function getAvailableEditors(): Promise<
-  ReadonlyArray<IFoundEditor<ExternalEditor>>
+  ReadonlyArray<IFoundEditor<string>>
 > {
-  const results: Array<IFoundEditor<ExternalEditor>> = []
-
-  const [
-    atomPath,
-    atomBetaPath,
-    atomNightlyPath,
-    codePath,
-    codeInsidersPath,
-    codiumPath,
-    sublimePath,
-    cfBuilderPath,
-    typoraPath,
-    slickeditPath,
-    webstormPath,
-    phpstormPath,
-    notepadPlusPlusPath,
-    riderPath,
-  ] = await Promise.all([
-    findApplication(ExternalEditor.Atom),
-    findApplication(ExternalEditor.AtomBeta),
-    findApplication(ExternalEditor.AtomNightly),
-    findApplication(ExternalEditor.VSCode),
-    findApplication(ExternalEditor.VSCodeInsiders),
-    findApplication(ExternalEditor.VSCodium),
-    findApplication(ExternalEditor.SublimeText),
-    findApplication(ExternalEditor.CFBuilder),
-    findApplication(ExternalEditor.Typora),
-    findApplication(ExternalEditor.SlickEdit),
-    findApplication(ExternalEditor.Webstorm),
-    findApplication(ExternalEditor.Phpstorm),
-    findApplication(ExternalEditor.NotepadPlusPlus),
-    findApplication(ExternalEditor.Rider),
-  ])
-
-  if (atomPath) {
-    results.push({
-      editor: ExternalEditor.Atom,
-      path: atomPath,
-      usesShell: true,
-    })
-  }
-
-  if (atomBetaPath) {
-    results.push({
-      editor: ExternalEditor.AtomBeta,
-      path: atomBetaPath,
-      usesShell: true,
-    })
-  }
+  const results: Array<IFoundEditor<string>> = []
 
-  if (atomNightlyPath) {
-    results.push({
-      editor: ExternalEditor.AtomNightly,
-      path: atomNightlyPath,
-      usesShell: true,
-    })
-  }
-
-  if (codePath) {
-    results.push({
-      editor: ExternalEditor.VSCode,
-      path: codePath,
-      usesShell: true,
-    })
-  }
-
-  if (codeInsidersPath) {
-    results.push({
-      editor: ExternalEditor.VSCodeInsiders,
-      path: codeInsidersPath,
-      usesShell: true,
-    })
-  }
-
-  if (codiumPath) {
-    results.push({
-      editor: ExternalEditor.VSCodium,
-      path: codiumPath,
-      usesShell: true,
-    })
-  }
-
-  if (sublimePath) {
-    results.push({
-      editor: ExternalEditor.SublimeText,
-      path: sublimePath,
-      usesShell: false,
-    })
-  }
-
-  if (cfBuilderPath) {
-    results.push({
-      editor: ExternalEditor.CFBuilder,
-      path: cfBuilderPath,
-      usesShell: false,
-    })
-  }
-
-  if (typoraPath) {
-    results.push({
-      editor: ExternalEditor.Typora,
-      path: typoraPath,
-      usesShell: false,
-    })
-  }
-
-  if (slickeditPath) {
-    results.push({
-      editor: ExternalEditor.SlickEdit,
-      path: slickeditPath,
-    })
-  }
-
-  if (webstormPath) {
-    results.push({
-      editor: ExternalEditor.Webstorm,
-      path: webstormPath,
-    })
-  }
+  const editorPaths = await Promise.all(
+    editors.map(editor =>
+      findApplication(editor).then(path => {
+        return { editor, path }
```

Comment:
Ad-hoc type + TypeScript compiler help = 😍 

But does it "feel" right for a TypeScript coder? 😳